From b9775c54ebf684d13dc1ef906242ccd8f3885ebb Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 21 Jun 2018 16:48:15 +0530
Subject: [PATCH 0375/1852] drm: xlnx: sdi: Fix VTC polarity for SD modes

commit 0d606d7e68f3030251b3769cb6d8a37ee626b5fd from
https://github.com/Xilinx/linux-xlnx.git

Fix the polarity flags set in VTC for SD mode by correcting the DRM
flags for the associated modes.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi_modes.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi_modes.h b/drivers/gpu/drm/xlnx/xlnx_sdi_modes.h
index d4b55ac1cbf5..1f606ef6b71f 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi_modes.h
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi_modes.h
@@ -40,14 +40,14 @@ static const struct xlnx_sdi_display_config xlnx_sdi_modes[] = {
 	/* SD: 720x480i@60Hz */
 	{{ DRM_MODE("720x480i", DRM_MODE_TYPE_DRIVER, 13500, 720, 739,
 		   801, 858, 0, 240, 244, 247, 262, 0,
-		   DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC |
 		   DRM_MODE_FLAG_INTERLACE | DRM_MODE_FLAG_DBLCLK),
 		   .vrefresh = 60, }, {0x7, 0x6},
 		   {0x81, 0x81, 0x81, 0x81, 0x81, 0x81} },
 	/* SD: 720x576i@50Hz */
 	{{ DRM_MODE("720x576i", DRM_MODE_TYPE_DRIVER, 13500, 720, 732,
 		   795, 864, 0, 288, 290, 293, 312, 0,
-		   DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC |
 		   DRM_MODE_FLAG_INTERLACE | DRM_MODE_FLAG_DBLCLK),
 		   .vrefresh = 50, }, {0x9, 0x9},
 		   {0x81, 0x81, 0x81, 0x81, 0x81, 0x81} },
-- 
2.31.1

